osu.Game/Screens/Select/BeatmapInfoWedgeV2.cs

```diff
@@ -0,0 +1,339 @@
+                        RelativeSizeAxes = Axes.Y,
+
+                        // By limiting the width we avoid this box showing up as an outline around the drawables that are on top of it.
+                        Width = colour_bar_width + corner_radius,
+                        Child = new Box { RelativeSizeAxes = Axes.Both }
+                    },
+                    new Container
+                    {
+                        // Applying the shear to this container and nesting the starCounter inside avoids
+                        // the deformation that occurs if the shear is applied to the starCounter whilst rotated
+                        Shear = -wedged_container_shear,
+                        X = -colour_bar_width / 2,
+                        Anchor = Anchor.CentreRight,
+                        Origin = Anchor.Centre,
+                        RelativeSizeAxes = Axes.Y,
+                        Width = colour_bar_width,
+                        Child = starCounter = new StarCounter
+                        {
+                            Rotation = (float)(Math.Atan(shear_width / WEDGE_HEIGHT) * (180 / Math.PI)),
+                            Colour = Colour4.Transparent,
+                            Anchor = Anchor.Centre,
+                            Origin = Anchor.Centre,
+                            Scale = new Vector2(0.35f),
+                            Direction = FillDirection.Vertical
+                        }
+                    }
+                }
+            };
+        }
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            ruleset.BindValueChanged(_ => updateDisplay());
+        }
+
+        private const double animation_duration = 600;
+
+        protected override void PopIn()
+        {
+            this.MoveToX(0, animation_duration, Easing.OutQuint);
+            this.FadeIn(200, Easing.In);
+        }
+
+        protected override void PopOut()
+        {
+            this.MoveToX(-150, animation_duration, Easing.OutQuint);
+            this.FadeOut(200, Easing.OutQuint);
+        }
+
+        private WorkingBeatmap? beatmap;
+
+        public WorkingBeatmap? Beatmap
+        {
+            get => beatmap;
+            set
+            {
+                if (beatmap == value) return;
+
+                beatmap = value;
+
+                updateDisplay();
+            }
+        }
+
+        private Container? loadingInfo;
+
+        private void updateDisplay()
+        {
+            Scheduler.AddOnce(() =>
+            {
+                if (beatmap == null)
+                {
+                    removeOldInfo();
+                    return;
+                }
+
+                LoadComponentAsync(loadingInfo = new Container
+                {
+                    Masking = true,
+                    // We offset this by the portion of the colour bar underneath we wish to show
+                    X = -colour_bar_width,
```

Comment:
This kinda stuff is SO dodgy I don't even know how it's working.

![osu Game Tests 2023-03-16 at 07 57 19](https://user-images.githubusercontent.com/191335/225551447-0b211a54-b784-44d3-9eb6-7a11fb76fd23.png)

I dunno.

Reply:
Super late reply. Irl stuff has had me busy, what exactly about it is the dodgy as hell?

Reply:
I believe this should use right padding instead of a negative X offset?

Reply:
Something liek that yes